### PR TITLE
Adding support for dbt 1.7

### PR DESCRIPTION
**Pull request by pgoslatara**
This PR adds support for dbt 1.7. Only point of difference from 1.6 is the introduction of a change to `dbt clean` in [#8469](https://github.com/dbt-labs/dbt-core/pull/8469) which prevents `dbt clean` from deleting directories outside the current project scope. To work around this, the environment variable `DBT_CLEAN_PROJECT_FILES_ONLY` is Monkeypatched. 

Additionally, `.gitignore` is added so that `tox` can be run locally without adding new files to any commits.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -70,3 +71,10 @@ deps =
     {[testenv]deps}
     dbt-core~=1.6.0
     dbt-postgres~=1.6.0
+
+[testenv:dbt17]
+description = run with dbt==1.7
+deps =
```
Comment (sweco) [inline tox.ini:77]: Can we set the environment variable here?

```suggestion
deps =
setenv =
    DBT_CLEAN_PROJECT_FILES_ONLY=false
```

File: tests/integration/test_init.py [hunk 2]
```diff
@@ -54,7 +54,10 @@ def setup_postgres(postgres_service):
 
 @pytest.fixture
 def setup_dbt(setup_postgres):
-    subprocess.run(["dbt", "clean", *DBT_ARGS], check=True)
+    with pytest.MonkeyPatch.context() as mp:
+        mp.setenv('DBT_CLEAN_PROJECT_FILES_ONLY', 'false')
+        subprocess.run(["dbt", "clean", *DBT_ARGS], check=True)
+
```
Comment (sweco) [inline tests/integration/test_init.py:60]: Can you please revert this and rather include the environment variable in the `tox.ini`? I'm just trying to get rid of the monkey-patching.

```suggestion
    subprocess.run(["dbt", "clean", *DBT_ARGS], check=True)
```

**Review by sweco — COMMENTED** [review]
Hey @pgoslatara, many thanks for the PR. 😊 Looks all good, I just thought that it would be worth removing the monkey-patching if possible.

File: tox.ini (hunk 1 above)
Comment (pgoslatara) [reply tox.ini:77]: Done! 10e59ebd21c541c145056a78be21416f63aecd40

**Pull request merged**